### PR TITLE
build: fix devtools dev server

### PR DESCRIPTION
**Pull request by crisbeto**
I ran into this after I rebased #50656 and tried to address the feedback. The devtools dev server wasn't building, because some transient dependencies weren't listed in the package.json.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -161,6 +161,18 @@
...
+    "@babel/plugin-proposal-dynamic-import": "^7.18.6",
+    "@babel/plugin-proposal-export-namespace-from": "^7.18.9",
+    "@babel/plugin-proposal-json-strings": "^7.18.6",
+    "@babel/plugin-proposal-logical-assignment-operators": "^7.20.7",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.18.6",
+    "@babel/plugin-proposal-numeric-separator": "^7.18.6",
+    "@babel/plugin-proposal-object-rest-spread": "^7.20.7",
+    "@babel/plugin-proposal-optional-catch-binding": "^7.18.6",
+    "@babel/plugin-proposal-optional-chaining": "^7.21.0",
+    "@babel/plugin-proposal-private-methods": "^7.18.6",
```
Comment (devversion) [inline package.json:175]: Wouldn't just be https://github.com/angular/dev-infra/blob/main/shared-scripts/angular-optimization/BUILD.bazel#L48 needed?

or what was the error?

Comment (crisbeto) [reply package.json:175]: It was throwing the following. Not sure if this is the right way to solve it, but it did get the build to pass.

```
TDERR:
could not find dependency '@babel/plugin-proposal-private-methods' of '@angular/build-tooling/node_modules/@babel/preset-env'
ERROR: /Projects/angular/WORKSPACE:79:13: fetching yarn_install rule //external:npm: Traceback (most recent call last):
	File "/private/var/tmp/_bazel/003b62d2e7b494adc222a58ec2b88b62/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 998, column 24, in _yarn_install_impl
		_create_build_files(repository_ctx, "yarn_install", node, repository_ctx.attr.yarn_lock, repository_ctx.attr.generate_local_modules_build_files)
	File "/private/var/tmp/_bazel/003b62d2e7b494adc222a58ec2b88b62/external/build_bazel_rules_nodejs/internal/npm_install/npm_install.bzl", line 509, column 13, in _create_build_files
		fail("generate_build_file.ts failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (result.stdout, result.stderr))
Error in fail: generate_build_file.ts failed: 
STDOUT:

STDERR:
could not find dependency '@babel/plugin-proposal-private-methods' of '@angular/build-tooling/node_modules/@babel/preset-env'
ERROR: no such package '@npm//@bazel/protractor': generate_build_file.ts failed: 
```

Comment (devversion) [reply package.json:175]: Oh that's seems like just the Yarn 1.x bug that will eventually be solvable when we migrate away from `rules_nodejs`. See: https://angular-team.slack.com/archives/C042EU9T5/p1653643888522619

Comment (crisbeto) [reply package.json:175]: In that case should we go with this fix for now? Otherwise it's difficult to work on the dev tools.

Comment (devversion) [reply package.json:175]: It's usually just having to `rm -rf nde_modules`. It's a bug that only surfaces when the Babel dependencies changed and `yarn` did not clear the older now-unused Babel deps. I think we should close this PR as it won't do anything

Comment (crisbeto) [reply package.json:175]: You're right, that fixed it. Closing.

**Pull request closed without merging**